lib/process-data-availability.js

```diff
@@ -236,205 +232,276 @@ const processDataAvailability = async (availabilityReport) => {
   const availabilityAggregatesByResource = {};
 
   //process fields
-  fields.forEach((field) => {
-    const { resourceName, fieldName } = field;
+  fields.forEach(field => {
+    const { resourceName, fieldName, parentResourceName = '' } = field;
 
-    const availability =
-      field?.frequency && resourceCounts[resourceName]
-        ? (1.0 * field.frequency) / resourceCounts[resourceName]
-        : 0.0;
+    const count = parentResourceName ? nestedResourcesCount[parentResourceName]?.[resourceName] : resourceCounts[resourceName];
+    const availability = field?.frequency && count ? (1.0 * field.frequency) / count : 0.0;
 
     //create template if not already present
     if (!availabilityAggregatesByResource[resourceName]) {
-      availabilityAggregatesByResource[resourceName] =
-        getAvailabilityAggregatesByResourceTemplate();
+      availabilityAggregatesByResource[resourceName] = getAvailabilityAggregatesByResourceTemplate();
+    }
+
+    if (parentResourceName) {
```

Comment:
Consider using a property for this instead.

`const isExpansion = parentResourceName && parentResourceName?.length;`